controllers/metal3machine_controller.go

```diff
@@ -156,21 +156,30 @@ func (r *Metal3MachineReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	}
 
 	// Check pause annotation on associated bmh (if any)
-	if !cluster.Spec.Paused {
+	if cluster.Spec.Paused != nil {
+		if !*cluster.Spec.Paused {
```

Comment:
This logic is a bit hard to understand for me, we had a chat offline. Lets fix the condition a bit

Reply:
improved it little bit please check